tests/test_sft_trainer.py

```diff
@@ -2183,3 +2185,201 @@ def test_handler(element, **kwargs):
+        ),
+        (
+            [
+                NESTFUL_DATA_INPUT_OUTPUT_JSONL,
+                TWITTER_COMPLAINTS_DATA_INPUT_OUTPUT_JSONL,
+            ],
+            DATA_CONFIG_MULTIPLE_DATASETS_ODM_YAML,
+            "GRADNORM",
+        ),
+    ],
+)
+def test_online_data_mixing_plugin_sample_training(
+    datafiles, datasetconfigname, reward_type
+):
+    """Ensure fms_acceleration_odm plugin does a sample training without failing"""
+    with tempfile.TemporaryDirectory() as tempdir:
+        data_formatting_args = copy.deepcopy(DATA_ARGS)
+
+        # set training_data_path and response_template to none
+        data_formatting_args.response_template = None
+        data_formatting_args.training_data_path = None
+
+        # add data_paths in data_config file
+        with tempfile.NamedTemporaryFile(
+            "w", delete=False, suffix=".yaml"
+        ) as temp_yaml_file:
+            with open(datasetconfigname, "r", encoding="utf-8") as f:
+                data = yaml.safe_load(f)
+                data["dataprocessor"]["odm"]["reward_type"] = reward_type
+                data["datasets"] = data["datasets"][:2]
+                for d, df in zip(data["datasets"], datafiles):
+                    d["data_paths"] = [df]
+                yaml.dump(data, temp_yaml_file)
+                data_formatting_args.data_config_path = temp_yaml_file.name
+
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+        train_args.logging_strategy = "steps"
+        train_args.max_steps = 2
+        train_args.eval_strategy = "steps"
+        train_args.eval_steps = 1
+
+        sft_trainer.train(MODEL_ARGS, data_formatting_args, train_args)
+
+        # validate full ft configs
+        _validate_training(tempdir)
+        _, checkpoint_path = _get_latest_checkpoint_trainer_state(tempdir)
+
+        # Load the model
+        loaded_model = TunedCausalLM.load(checkpoint_path, MODEL_NAME)
+
+        # Run inference on the text
+        output_inference = loaded_model.run(
+            "### Text: @NortonSupport Thanks much.\n\n### Label:", max_new_tokens=50
+        )
+        assert len(output_inference) > 0
+        assert "### Text: @NortonSupport Thanks much.\n\n### Label:" in output_inference
+
+        output_inference = loaded_model.run(
+            "It takes 10 days for digging a trench of 100 m long, "
+            "50 m broad and 10 m deep. What length of trench,\n25 m broad and 15 m "
+            "deep can be dug in 30 days ?",
+            max_new_tokens=50,
+        )
+        assert len(output_inference) > 0
+        assert (
+            "It takes 10 days for digging a trench of 100 m long, 50 m broad and 10 m deep. "
+            "What length of trench,\n25 m broad and 15 m deep can be dug in 30 days ?"
+            in output_inference
+        ), f"{output_inference} does not include the prompt"
+
+
+@pytest.mark.skipif(
+    not is_fms_accelerate_available(plugins="odm"),
```

Comment:
Are we enabling installing `odm` plugin by default in the `tox.ini` file?
if not i am assmuming these tests are run separately

Reply:
No we dont want to install the odm plugin by default. Yes I have run these tests on my end:

Check the screenshots below:

<img width="838" height="392" alt="Screenshot 2025-10-07 at 5 48 48 PM" src="https://github.com/user-attachments/assets/4a9e4fc6-6eb1-4d1e-82ec-7b36fa6966c9" />

<img width="857" height="443" alt="Screenshot 2025-10-07 at 5 50 55 PM" src="https://github.com/user-attachments/assets/b53f2d22-63d8-4175-898f-c6f307035368" />